haystack_experimental/components/preprocessors/embedding_based_document_splitter.py

```diff
@@ -0,0 +1,402 @@
+        "Completely different topic. The same completely different topic."
+    )
+
+    embedder = SentenceTransformersDocumentEmbedder()
+
+    splitter = EmbeddingBasedDocumentSplitter(
+        document_embedder=embedder,
+        sentences_per_group=2,
+        percentile=0.95,
+        min_length=50,
+        max_length=1000
+    )
+    splitter.warm_up()
+    result = splitter.run(documents=[doc])
+    ```
+    """
+
+    def __init__(
+        self,
+        *,
+        document_embedder: DocumentEmbedder,
+        sentences_per_group: int = 3,
+        percentile: float = 0.95,
+        min_length: int = 50,
+        max_length: int = 1000,
+        language: Language = "en",
+        use_split_rules: bool = True,
+        extend_abbreviations: bool = True,
+    ):
+        """
+        Initialize EmbeddingBasedDocumentSplitter.
+
+        :param document_embedder: The DocumentEmbedder to use for calculating embeddings.
+        :param sentences_per_group: Number of sentences to group together before embedding.
+        :param percentile: Percentile threshold for cosine distance. Distances above this percentile
+            are treated as break points.
+        :param min_length: Minimum length of splits in characters. Splits below this length will be merged.
+        :param max_length: Maximum length of splits in characters. Splits above this length will be recursively split.
+        :param language: Language for sentence tokenization.
+        :param use_split_rules: Whether to use additional split rules for sentence tokenization.
+        :param extend_abbreviations: Whether to extend NLTK abbreviations.
+        """
+        self.document_embedder = document_embedder
+        self.sentences_per_group = sentences_per_group
+        self.percentile = percentile
+        self.min_length = min_length
+        self.max_length = max_length
+        self.language = language
+        self.use_split_rules = use_split_rules
+        self.extend_abbreviations = extend_abbreviations
+
+        self._init_validation()
+        self.sentence_splitter: Optional[SentenceSplitter] = None
+        self._is_warmed_up = False
+
+    def _init_validation(self) -> None:
+        """
+        Validates initialization parameters.
+        """
+        if self.sentences_per_group <= 0:
+            raise ValueError("sentences_per_group must be greater than 0.")
+
+        if not 0.0 <= self.percentile <= 1.0:
+            raise ValueError("percentile must be between 0.0 and 1.0.")
+
+        if self.min_length < 0:
+            raise ValueError("min_length must be greater than or equal to 0.")
+
+        if self.max_length <= self.min_length:
+            raise ValueError("max_length must be greater than min_length.")
+
+    def warm_up(self):
+        """
+        Warm up the component by initializing the sentence splitter.
+        """
+        nltk_imports.check()
+        self.sentence_splitter = SentenceSplitter(
```

Comment:
Not entirely sure if the `nltk_imports.check()` is needed here since it will be called by `SentenceSplitter` when it is initialized.